src/frequenz/sdk/microgrid/_data_pipeline.py

```diff
@@ -119,29 +118,22 @@ def __init__(
         self._grid: Grid | None = None
         self._ev_charger_pools: dict[frozenset[int], EVChargerPool] = {}
         self._battery_pools: dict[frozenset[int], BatteryPoolReferenceStore] = {}
-        self._frequency_pool: dict[int, GridFrequency] = {}
+        self._frequency_instance: GridFrequency
 
-    def frequency(self, component: Component | None = None) -> GridFrequency:
+    def frequency(self) -> GridFrequency:
         """Fetch the grid frequency for the microgrid.
 
-        Args:
-            component: The component to use when fetching the grid frequency.  If None,
-                the component will be fetched from the registry.
-
         Returns:
-            A GridFrequency instance.
+            The GridFrequency instance.
         """
-        if component is None:
-            component = GridFrequency.find_frequency_component()
+        if not hasattr(self, "_frequency_instance"):
+            self._frequency_instance = GridFrequency(
+                self._data_sourcing_request_sender(),
+                self._channel_registry,
+                GridFrequency.find_frequency_source(),
```

Comment:
Also, since now we don't need the `component_id` to lookup the cache to see if the object was already created, any reason not to at least make the `source` optional in `GridFrequency` and find the best component automatically if the `source` is omitted?